include/ros1_bridge/factory.hpp

```diff
@@ -241,34 +241,37 @@ class ServiceFactory : public ServiceFactoryInterface
   using ROS2Response = typename ROS2_T::Response;
 
   void forward_2_to_1(
-    ros::ServiceClient client, const std::shared_ptr<rmw_request_id_t>,
+    ros::ServiceClient & client, rclcpp::Logger logger, const std::shared_ptr<rmw_request_id_t>,
     const std::shared_ptr<ROS2Request> request, std::shared_ptr<ROS2Response> response)
   {
     ROS1_T srv;
+    auto service_name = client.getService();
     translate_2_to_1(*request, srv.request);
     if (client.call(srv)) {
       translate_1_to_2(srv.response, *response);
     } else {
-      throw std::runtime_error("Failed to get response from ROS service");
+      throw std::runtime_error("Failed to get response from ROS1 service " + service_name);
     }
   }
 
   bool forward_1_to_2(
-    rclcpp::ClientBase::SharedPtr cli,
+    rclcpp::ClientBase::SharedPtr cli, rclcpp::Logger logger,
     const ROS1Request & request1, ROS1Response & response1)
   {
+    auto service_name = std::string(cli->get_service_name());
     auto client = std::dynamic_pointer_cast<rclcpp::Client<ROS2_T>>(cli);
     if (!client) {
-      fprintf(stderr, "Failed to get the client.\n");
+      RCLCPP_ERROR(logger, "Failed to get ROS2 client " + service_name);
       return false;
     }
     auto request2 = std::make_shared<ROS2Request>();
     translate_1_to_2(request1, *request2);
-    while (!client->wait_for_service(std::chrono::seconds(1))) {
+    while (!client->wait_for_service(std::chrono::seconds(5))) {
```

Comment:
This change is also unrelated to the logging. Why was the timeout changed?

Reply:
Purged, thanks